dask/dataframe/shuffle.py

```diff
@@ -791,11 +791,8 @@ def set_partitions_pre(s, divisions, ascending=True, na_position="last"):
         else:
             partitions = len(divisions) - divisions.searchsorted(s, side="right") - 1
     except TypeError:
-        # When `searchsorted` fails with `TypeError`, it may be
-        # caused by nulls in `s`. Try again with the null-values
-        # explicitly mapped to the first partition.
+        # `searchsorted` fails if `s` contains nulls and strings
         partitions = np.empty(len(s), dtype="int32")
-        partitions[s.isna()] = 0
```

Comment:
Why is this being removed ?

Reply:
We end up reassigning to `partitions[s.isna()]` a few lines down with

```
    partitions[s.isna().values] = len(divisions) - 2 if na_position == "last" else 0
```

Regardless of if we enter the `except` block

Reply:
I *think* this logic is handled in the last line before `return`.